lametro/management/commands/generate_tag_analytics.py

```diff
@@ -1,34 +1,97 @@
-            writer.writerow(
-                ('File ID',
-                 'Identifier'
-                 'Title',
-                 'Last Action Date',
-                 'Tag Classification',
-                 'Tag')
-            )
-
-            for bill in tqdm(LAMetroBill.objects.all()):
-                for tag in bill.rich_topics:
-                    writer.writerow(
-                        (bill.board_report.id,
-                         bill.identifier,
-                         bill.friendly_name,
-                         bill.last_action_date,
-                         tag.classification,
-                         tag.name)
-                    )
+        csv_string = self.generate_tag_analytics()
+        date = datetime.today().strftime('%m_%d_%y')
+        output_file_name = f'{date}_tag_analytics.csv'
+
+
+        file_metadata = {
+            'name': output_file_name,
+            'parents': [settings.REMOTE_ANALYTICS_FOLDER]
+        }
+        drive_service = self.get_google_drive()
+
+        if not drive_service:
+            return
+
+        media = MediaIoBaseUpload(csv_string, mimetype='text/csv')
+        drive_service.files().create(
+                body=file_metadata,
+                media_body=media,
+                fields='id'
+            ).execute()
+
+        self.stdout.write(
+            self.style.SUCCESS(f'Successfully uploaded {output_file_name}!')
+        )
+
+    def generate_tag_analytics(self):
+        """Returns a CSV-formatted byte stream of the tag analytics"""
+
+        csv_string = StringIO()
+        writer = csv.writer(csv_string, delimiter=',')
+        writer.writerow(
+            ('File ID',
+             'Identifier'
+             'Title',
+             'File Name',
+             'Last Action Date',
+             'Tag Classification',
+             'Tag')
+        )
+
+        self.stdout.write('\nGenerating tag analytics...')
+        for bill in tqdm(LAMetroBill.objects.all()):
+            for tag in bill.rich_topics:
+                writer.writerow(
+                    (bill.board_report.id,
+                     bill.identifier,
+                     bill.friendly_name,
+                     bill.last_action_date,
+                     tag.classification,
+                     tag.name)
+                )
+
+        return csv_string
+
+    def get_google_drive(self):
+        """Authenticates a service account and returns a Google Drive object."""
+
+        SCOPES = [
+            'https://www.googleapis.com/auth/drive',
+        ]
+
+        try:
+            credentials = service_account.Credentials.from_service_account_file(
+                    settings.SERVICE_ACCOUNT_KEY_PATH
+                ).with_scopes(SCOPES)
+        except FileNotFoundError:
+            self.stdout.write(self.style.ERROR('Service account key file not found.'))
+            return
+
+        if not credentials:
+            self.stdout.write(self.style.ERROR('Could not generate credentials!'))
```

Comment:
Can you say more about logging rather than raising exceptions on these conditions?

Reply:
I did think about raising exceptions instead but I thought that since I wasn't going to be including logic to handle the exception (and the script will just abort), it would suffice to just display the error. 

Is it preferable to raise exceptions in situations like this?

Reply:
My rule of thumb is that if an exception prevents a program from completing its intended work, it should raise an exception. In this case, we'll be running this from Airflow, and we'll definitely want the indicator that this command failed so we know we have a problem to fix.

Reply:
Ok cool. I'll add in exceptions.

Reply:
@hancush And would it still be helpful to handle the exception in a try/except block by printing to stdout with the error info?